### PR TITLE
RavenDB-15522 - Migration of revisions without a document from v3.x

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-15522

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 1]
```diff
@@ -39,8 +39,7 @@ public class DatabaseSmuggler
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static bool IsPreV4Revision(BuildVersionType buildType, string id, Document document)
         {
-            if (buildType == BuildVersionType.V3 == false)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: I do not see a good reason for changing this. Can we revert?

Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: done

**Comment by ppekrol** [comment]
@grisha-kotler can you check failing tests?

**Comment by grisha-kotler** [comment]
done

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs [hunk 2]
```diff
@@ -988,15 +991,42 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
...
+
+                        document.Flags |= DocumentFlags.HasRevisions;
                         _database.DocumentsStorage.RevisionsStorage.Put(context, newId, document.Data, document.Flags,
                             document.NonPersistentFlags, document.ChangeVector, document.LastModified.Ticks);
+
+                        if (parentDocument != null)
+                        {
+                            // the change vector of the document must be identical to the one of the last revision
+                            databaseChangeVector = ChangeVectorUtils.MergeVectors(databaseChangeVector, document.ChangeVector);
+                            parentDocument.Data = parentDocument.Data.Clone(context);
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: Don't we need to dispose the old data?

```
using (var old = parentDocument.Data)
{
    parentDocument.Data = parentDocument.Data.Clone(context);
}
```

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 3]
```diff
@@ -648,6 +658,9 @@ private SmugglerProgressBase.Counts ProcessDocuments(SmugglerResult result, Buil
                 TryHandleLegacyDocumentTombstones(legacyIdsToDelete, actions, result);
             }
 
+            if (buildType == BuildVersionType.V3 && result.RevisionDocuments.ReadCount > 0)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs:664]: Move it to this block:
```
            if (ensureStepsProcessed)
            {
                counts.Processed = true;

                if (counts is SmugglerProgressBase.CountsWithLastEtagAndAttachments countsWithEtagAndAttachments)
                {
                    countsWithEtagAndAttachments.Attachments.Processed = true;

                    switch (type)
                    {
                        case DatabaseItemType.Documents:
                        case DatabaseItemType.RevisionDocuments:
                            countsWithEtagAndAttachments.Attachments.Skipped = _options.OperateOnTypes.HasFlag(DatabaseItemType.Attachments) == false;
                            break;
                    }
                }
            }
```

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: done

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs:664]: done

**Pull request merged**